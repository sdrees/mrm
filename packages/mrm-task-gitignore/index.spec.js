jest.mock('fs');
jest.mock('mrm-core/src/util/log', () => ({
	added: jest.fn(),
}));

const { getConfigGetter } = require('mrm');
const vol = require('memfs').vol;
const task = require('./index');

afterEach(() => vol.reset());

it('should add .gitignore', () => {
	vol.fromJSON();

	task(getConfigGetter({}));

	expect(vol.toJSON()).toMatchSnapshot();
});

it('should add package-lock.json, if yarn.lock exists', () => {
	vol.fromJSON({
		'/yarn.lock': '',
	});

	task(getConfigGetter({}));

	expect(vol.toJSON()).toMatchSnapshot();
});

it('should add yarn.lock, if package-lock.json exists', () => {
	vol.fromJSON({
		'/package-lock.json': '',
	});

	task(getConfigGetter({}));

	expect(vol.toJSON()).toMatchSnapshot();
});
