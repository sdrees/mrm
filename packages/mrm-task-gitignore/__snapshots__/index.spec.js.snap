// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add .gitignore 1`] = `
Object {
  "/.gitignore": "node_modules/
.DS_Store
Thumbs.db
.idea/
.vscode/
*.sublime-project
*.sublime-workspace
*.log",
}
`;

exports[`should add package-lock.json, if yarn.lock exists 1`] = `
Object {
  "/.gitignore": "node_modules/
.DS_Store
Thumbs.db
.idea/
.vscode/
*.sublime-project
*.sublime-workspace
*.log
package-lock.json",
  "/yarn.lock": "",
}
`;

exports[`should add yarn.lock, if package-lock.json exists 1`] = `
Object {
  "/.gitignore": "node_modules/
.DS_Store
Thumbs.db
.idea/
.vscode/
*.sublime-project
*.sublime-workspace
*.log
yarn.lock",
  "/package-lock.json": "",
}
`;
